src/main/io/ledstrip.c

```diff
@@ -1057,6 +1067,67 @@ static void applyRainbowLayer(bool updateNow, timeUs_t *timer)
     }
 }
 
+typedef struct {
+    uint16_t ttl, age, width;
+    float center, speedFactor;
+    bool goingLeft, alive;
+} auroraWave_t;
+
+static void initAuroraWave(auroraWave_t *wave, int segmentLength) {
+    wave->ttl = rand() % 1001 + 500;
+    wave->age = 0;
+    wave->width = rand() % (segmentLength / ledStripConfig()->w_width_factor_aurora) + (segmentLength / 20);
+    wave->center = rand() / (float)RAND_MAX * segmentLength;
+    wave->speedFactor = (rand() % 21 + 10) / 2550.0 * ledStripConfig()->w_max_speed_aurora;
+    wave->goingLeft = rand() % 2;
+    wave->alive = true;
+}
+
+static void updateAuroraWave(auroraWave_t *wave, int segmentLength) {
+    wave->center += wave->goingLeft ? -wave->speedFactor : wave->speedFactor;
+    wave->alive = ++wave->age <= wave->ttl && wave->center + wave->width >= 0 && wave->center - wave->width <= segmentLength;
+}
+
+static int brightnessForAuroraIndex(const auroraWave_t *wave, uint8_t ledIndex) {
+    float offset = fabsf(ledIndex - wave->center);
+    return (wave->alive && offset <= wave->width) 
+        ? (int)((1 - offset / wave->width) * (wave->ttl - wave->age) / wave->ttl * 255) 
+        : 0;
+}
+
+static void applyAuroraLayer(bool updateNow, timeUs_t *timer) {
+    static auroraWave_t *waves = NULL;
+    static uint8_t waveCount = 0;
+
+    if (!waves) {
+        waveCount = ledStripConfig()->w_max_count_aurora;
+        if (!(waves = malloc(waveCount * sizeof(auroraWave_t)))) return;
+        memset(waves, 0, waveCount * sizeof(auroraWave_t)); // Initialisiert alle Wellen
```

Comment:
please use static initialization on the stack instead of dynamic initialization on the heap. It's a safety issue because we are dealing with aircraft.